jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/DiagnosticUtils.java

```diff
@@ -348,6 +387,130 @@ private static DiagnosticProperties doDump(String diagnosticCommand) {
+		switch (timeUnit) {
+		case "s":
+			timeInMilli = TimeUnit.SECONDS.toMillis(time);
+			break;
+		case "m":
+			timeInMilli = TimeUnit.MINUTES.toMillis(time);
+			break;
+		case "h":
+			timeInMilli = TimeUnit.HOURS.toMillis(time);
+			break;
+		case "d":
+			timeInMilli = TimeUnit.DAYS.toMillis(time);
+			break;
+		default:
+			// no unit or unrecognized unit, assume milliseconds
+			timeInMilli = time;
+			break;
+		}
+		return Long.valueOf(timeInMilli);
+	}
+
+	private static Long parseTimeParameter(String paramName, String[] parameters) {
+		for (String param : parameters) {
+			if (param.startsWith(paramName + "=")) {
+				int valueStart = param.indexOf("=");
+				if (valueStart != -1) {
+					return convertToMilliseconds(param.substring(valueStart + 1));
+				}
+			}
+		}
+		return null;
+	}
+
+	private static String parseStringParameter(String paramName, String[] parameters, String defaultValue) {
+		for (String param : parameters) {
+			if (param.startsWith(paramName + "=")) {
+				int valueStart = param.indexOf("=");
+				if (valueStart != -1) {
+					return param.substring(valueStart + 1);
+				}
+			}
+		}
+		return defaultValue;
+	}
+
+	private static DiagnosticProperties doJFR(String diagnosticCommand) {
+		DiagnosticProperties result = null;
+
+		// split the command and arguments
+		String[] parts = diagnosticCommand.split(DIAGNOSTICS_OPTION_SEPARATOR);
+		IPC.logMessage("doJFR: ", diagnosticCommand);
+
+		// ensure there's at least one part for the command
+		if (parts.length == 0) {
+			return DiagnosticProperties.makeErrorProperties("Error: No JFR command specified");
+		}
+
+		String command = parts[0].trim();
+		String[] parameters = Arrays.copyOfRange(parts, 1, parts.length);
+
+		String fileName = parseStringParameter("filename", parameters, null);
+		IPC.logMessage("doJFR: filename = ", fileName);
+		boolean setFileName = (fileName != null) && !fileName.isEmpty();
+		if (setFileName) {
+			if (!VM.setJFRRecordingFileName(fileName)) {
+				return DiagnosticProperties.makeErrorProperties("setJFRRecordingFileName failed");
+			} else {
+				jfrRecordingFileName = fileName;
+			}
+		}
+
+		try {
+			if (command.equalsIgnoreCase(DIAGNOSTICS_JFR_START)) {
+				VM.startJFR();
+				Long duration = parseTimeParameter("duration", parameters);
+				IPC.logMessage("doJFR: ", diagnosticCommand);
+				if (duration != null) {
+					long delay = duration.longValue();
+					if (delay > 0) {
+						Timer timer = new Timer();
+						TimerTask jfrDumpTask = new TimerTask() {
+							public void run() {
+								VM.triggerExecutionSample();
+								VM.stopJFR();
+							}
+						};
+						timer.schedule(jfrDumpTask, duration.longValue());
+					} else {
+						// the recording is on until next JFR command
+					}
+				}
+				result = DiagnosticProperties.makeStringResult("Start JFR recording to " + jfrRecordingFileName);
+			} else if (command.equalsIgnoreCase(DIAGNOSTICS_JFR_STOP)) {
+				VM.triggerExecutionSample();
```

Comment:
You dont need to call `VM.triggerExecutionSample();` thats just for testing purposes

Reply:
To clarify the intended behaviours of `jcmd` commands:
* `JFR.start` - initializeJFR() and start jfrStartSamplingThread(), the recording file is created with the default name, the file might contain some thread info at the beginning of JFR start time;
* `JFR.dump` - walk all Java threads and flush the buffer, then flush global memory out, the content will be appended to the existing recording file. 
 Does `triggerExecutionSample()` write the same contents to the file as a `JFR.dump`? 
 Does the recording file created by `JFR.dump` have `JFR_HEADER_SPECIALFLAGS_LAST_CHUNK`?
 Assuming multiple `JFR.dump` is possible, are the contents to be appended to the file created earlier?
* `JFR.stop` - tearDownJFR() walks all Java threads, flushes the buffer, and then flushes global memory out to the recording file, this step is similar to a `JFR.dump`. Afterward, unregister the hooks and free all local buffers.

Does this make sense?

Reply:
> JFR.start - initializeJFR() and start jfrStartSamplingThread(), the recording file is created with the default name, the file might contain some thread info at the beginning of JFR start time;

Yes, if the user specifies a name then that one should be used.

> JFR.dump - walk all Java threads and flush the buffer, then flush global memory out, the content will be appended to the existing recording file.

Yes

> Does triggerExecutionSample() write the same contents to the file as a JFR.dump?

No, all it does is generate an event on all threads. Its really for testing purposes, not needed at all here.

> Does the recording file created by JFR.dump have JFR_HEADER_SPECIALFLAGS_LAST_CHUNK?

Its simply a way to say that this will be the last chunk, so really it should be set on stop.

> Assuming multiple JFR.dump is possible, are the contents to be appended to the file created earlier?

Dump should just flush the buffers andn append it out to the file. If its done multiple times it just updates the file.

> JFR.stop - tearDownJFR() walks all Java threads, flushes the buffer, and then flushes global memory out to the recording file, this step is similar to a JFR.dump. Afterward, unregister the hooks and free all local buffers.

Yes, stop basiclly does a dump, then it stops the recording.

Reply:
Removed all `triggerExecutionSample()` usages.